src/vcpkg/base/cofffilereader.cpp

```diff
@@ -41,12 +41,407 @@ namespace vcpkg
+        uint32_t Catalog;
+        uint16_t CatalogOffset;
+        uint16_t Reserved;
+    };
+
+    struct ImageLoadConfigDirectory32
+    {
+        uint32_t Size;
+        uint32_t TimeDateStamp;
+        uint16_t MajorVersion;
+        uint16_t MinorVersion;
+        uint32_t GlobalFlagsClear;
+        uint32_t GlobalFlagsSet;
+        uint32_t CriticalSectionDefaultTimeout;
+        uint32_t DeCommitFreeBlockThreshold;
+        uint32_t DeCommitTotalFreeThreshold;
+        uint32_t LockPrefixTable;
+        uint32_t MaximumAllocationSize;
+        uint32_t VirtualMemoryThreshold;
+        uint32_t ProcessHeapFlags;
+        uint32_t ProcessAffinityMask;
+        uint16_t CSDVersion;
+        uint16_t DependentLoadFlags;
+        uint32_t EditList;
+        uint32_t SecurityCookie;
+        uint32_t SEHandlerTable;
+        uint32_t SEHandlerCount;
+        uint32_t GuardCFCheckFunctionPointer;
+        uint32_t GuardCFDispatchFunctionPointer;
+        uint32_t GuardCFFunctionTable;
+        uint32_t GuardCFFunctionCount;
+        uint32_t GuardFlags;
+        ImageLoadConfigCodeIntegrity CodeIntegrity;
+        uint32_t GuardAddressTakenIatEntryTable;
+        uint32_t GuardAddressTakenIatEntryCount;
+        uint32_t GuardLongJumpTargetTable;
+        uint32_t GuardLongJumpTargetCount;
+        uint32_t DynamicValueRelocTable;
+        uint32_t CHPEMetadataPointer;
+    };
+
+    struct ImageLoadConfigDirectory64
+    {
+        uint32_t Size;
+        uint32_t TimeDateStamp;
+        uint16_t MajorVersion;
+        uint16_t MinorVersion;
+        uint32_t GlobalFlagsClear;
+        uint32_t GlobalFlagsSet;
+        uint32_t CriticalSectionDefaultTimeout;
+        uint64_t DeCommitFreeBlockThreshold;
+        uint64_t DeCommitTotalFreeThreshold;
+        uint64_t LockPrefixTable;
+        uint64_t MaximumAllocationSize;
+        uint64_t VirtualMemoryThreshold;
+        uint64_t ProcessAffinityMask;
+        uint32_t ProcessHeapFlags;
+        uint16_t CSDVersion;
+        uint16_t DependentLoadFlags;
+        uint64_t EditList;
+        uint64_t SecurityCookie;
+        uint64_t SEHandlerTable;
+        uint64_t SEHandlerCount;
+        uint64_t GuardCFCheckFunctionPointer;
+        uint64_t GuardCFDispatchFunctionPointer;
+        uint64_t GuardCFFunctionTable;
+        uint64_t GuardCFFunctionCount;
+        uint32_t GuardFlags;
+        ImageLoadConfigCodeIntegrity CodeIntegrity;
+        uint64_t GuardAddressTakenIatEntryTable;
+        uint64_t GuardAddressTakenIatEntryCount;
+        uint64_t GuardLongJumpTargetTable;
+        uint64_t GuardLongJumpTargetCount;
+        uint64_t DynamicValueRelocTable;
+        uint64_t CHPEMetadataPointer;
+    };
+
+    struct LoadedDll
+    {
+        CoffFileHeader coff_header;
+        CommonPEOptionalHeaders common_optional_headers;
+        union
+        {
+            UniquePEOptionalHeaders pe_headers;
+            UniquePEPlusOptionalHeaders pe_plus_headers;
+        };
+
```

Comment:
I think it would be better to read the load config in the ctor so that the magic number check for PE vs PE+ was in one place instead of duplicated in `check_load_config_dll_arm64ec` but that preference is minor...